### PR TITLE
AG-13334 - Move module mappings example to markdoc component

### PR DESCRIPTION
**Pull request by taktran**

### CODE REVIEW

File: documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        AllEnterpriseModule.dependsOn?.forEach(calcDependencies);
+    }, [dependencies]);
+
+    return (
+        <div style={{ height: '600px' }}>
+            <AgGridReact
+                defaultColDef={{
+                    flex: 1,
+                }}
+                columnDefs={[{ field: 'moduleName' }]}
```
Comment (peterjrreynolds) [inline documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx]: memoize all the col defs props

File: documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                }}
+                rowData={modules.groups}
+                treeData={true}
+                treeDataChildrenField="children"
+                getRowId={(params: GetRowIdParams) => params.data.name}
+                rowSelection={{
+                    mode: 'multiRow',
+                    groupSelects: 'descendants',
+                }}
+                onRowSelected={(event: RowSelectedEvent) => {
```
Comment (peterjrreynolds) [inline documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx]: `useCallback`

File: documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx (hunk 1 above)
Comment (taktran) [reply documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx]: Good catch - have updated.

File: documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx (hunk 2 above)
Comment (taktran) [reply documentation/ag-grid-docs/src/components/module-mappings/ModuleMappings.tsx]: Same

**Review by markdurrant — APPROVED** [review]
LGTM 👍

**Pull request merged**